geth/node/node.go

```diff
@@ -90,9 +90,11 @@ func MakeNode(config *params.NodeConfig) (*node.Node, error) {
 		return nil, ErrNodeMakeFailure
 	}
 
-	// start Ethereum service
-	if err := activateEthService(stack, config); err != nil {
-		return nil, fmt.Errorf("%v: %v", ErrEthServiceRegistrationFailure, err)
+	// start Ethereum service if we are not expected to use an upstream server.
+	if !config.UpstreamConfig.Enabled {
```

Comment:
Hm, let's log that `Blockchain synchronization is switched off due to UpstreamConfig.Enabled=true, RPC requests will be proxied to %s` or something like that.